src/main/resources/api/service_openapi.yaml

```diff
@@ -466,8 +466,7 @@ components:
         referenceType:
           $ref: '#/components/schemas/ReferenceTypeEnum'
         reference:
-          description: The serialized object containing details specific to the referenceType. For instance, TDR snapshots must be serialized json.
-          type: string
+          $ref: '#/components/schemas/DataRepoSnapshot'
```

Comment:
changing this to an actual object from a string representation of an object will impact services upstream, like rawls

Reply:
Yes, but the actual transmitted JSON will be the same, so while we'll need to make that change in Rawls when we publish and are using the new client, it'll be fine on release.

Reply:
is it true that the transmitted value is the same? I thought I saw some string-encoding going on somewhere. I am not certain of this.

Reply:
@zloery any insight?

Reply:
This actually made me curious so I went and checked - DA is right, this does change the transmitted value (and it's mutually unintelligible between the two versions).

A call to the existing system looks like
`curl -X POST "https://workspace.dsde-dev.broadinstitute.org/api/workspaces/v1/3fa85f64-5717-4562-b3fc-2c963f66afa6/datareferences" -H  "accept: */*" -H  "Authorization: Bearer ___" -H  "Content-Type: application/json" -d "{\"name\":\"snapshotName\",\"referenceType\":\"DATA_REPO_SNAPSHOT\",\"reference\":\"{\\\"instanceName\\\": \\\"terra-dev\\\", \\\"snapshot\\\": \\\"93dc1e76-8f1c-4949-8f9b-07a087f3ce7b\\\"}\",\"cloningInstructions\":\"COPY_NOTHING\"}"`

A call to the new system looks like
`curl -X POST "http://localhost:8080/api/workspaces/v1/3fa85f64-5717-4562-b3fc-2c963f66afa6/datareferences" -H  "accept: */*" -H  "Authorization: Bearer ___" -H  "Content-Type: application/json" -d "{\"name\":\"snapshotName\",\"referenceType\":\"DATA_REPO_SNAPSHOT\",\"reference\":{\"instanceName\":\"terra-dev\",\"snapshot\":\"93dc1e76-8f1c-4949-8f9b-07a087f3ce7b\"},\"cloningInstructions\":\"COPY_NOTHING\"}"`

The string encoding on the snapshot actually does change, so we'll have to update Rawls after all